src/rfquack_radio.h

```diff
@@ -29,89 +29,94 @@
+
+#ifndef RFQUACK_RADIO_RX_QUEUE_LEN
+#define RFQUACK_RADIO_RX_QUEUE_LEN RFQUACK_RADIO_RX_QUEUE_LEN_DEFAULT
+#endif
 
 /*****************************************************************************
  * Variables
  *****************************************************************************/
 Queue rfquack_rx_q(sizeof(rfquack_Packet), RFQUACK_RADIO_RX_QUEUE_LEN, FIFO,
                    true);
 
-/* Radio instance */
-RFQRadio rfquack_rf(RFQUACK_RADIO_PIN_CS, RFQUACK_RADIO_PIN_RST,
-                    RFQUACK_RADIO_PIN_IRQ);
+/*
+RFQRadio *rfquack_rf = new RFQnRF24(
+  new nRF24(
+    new Module(RFQUACK_RADIO_PIN_CS, RFQUACK_RADIO_PIN_IRQ, RFQUACK_RADIO_PIN_IRQ1)
+  )
+);
+ */
+RFQRadio *rfquack_rf;
+
 
 /**
  * @brief Changes the radio state in the radio driver.
  */
 void rfquack_update_mode() {
   if (rfq.mode == rfquack_Mode_IDLE) {
-    rfquack_rf.setModeIdle();
-#ifdef RFQUACK_LOG_ENABLED
-    Log.trace("Radio in IDLE mode");
-#endif
+    int16_t result = rfquack_rf->begin();
+    RFQUACK_LOG_TRACE("Radio in IDLE mode, resultCode=%d", result)
   }
 
   if (rfq.mode == rfquack_Mode_REPEAT || rfq.mode == rfquack_Mode_RX) {
-    rfquack_rf.setModeRx();
-#ifdef RFQUACK_LOG_ENABLED
-    Log.trace("Radio in RX mode");
-#endif
+    int16_t result = rfquack_rf->receiveMode();
+    RFQUACK_LOG_TRACE("Radio in RECEIVE mode, resultCode=%d", result)
   }
 
-  // never set TX mode, it happens automatically
+  if (rfq.mode == rfquack_Mode_TX) {
+    int16_t result = rfquack_rf->transmitMode();
+    RFQUACK_LOG_TRACE("Radio in TRANSMIT mode, resultCode=%d", result)
+  }
 }
 
 /**
  * @brief Changes preamble length in the radio driver.
  */
 void rfquack_update_preamble() {
-  rfquack_rf.setPreambleLength(rfq.modemConfig.preambleLen);
-  Log.trace("Preamble length:     %d bytes", rfq.modemConfig.preambleLen);
+  int16_t result = rfquack_rf->setPreambleLength(rfq.modemConfig.preambleLen);
+  RFQUACK_LOG_TRACE("Preamble length:     %d bytes, resultCode=%d", rfq.modemConfig.preambleLen, result)
 }
 
 /**
  * @brief Changes frequency in the radio driver.
  */
 void rfquack_update_frequency() {
-  if (!rfquack_rf.setFrequency(rfq.modemConfig.carrierFreq)) {
-    Log.error("⚠️  Set frequency failed");
-    return;
-  }
+  int16_t result = rfquack_rf->setFrequency(rfq.modemConfig.carrierFreq);
+  RFQUACK_LOG_TRACE("Frequency :     %f Mhz (?), resultCode=%d", rfq.modemConfig.carrierFreq, result)
 }
 
 /**
  * @brief Changes TX power in the radio driver
  */
 void rfquack_update_tx_power() {
-  rfquack_rf.setTxPower(rfq.modemConfig.txPower
-#ifdef RFQUACK_RADIO_SET_HIGHPOWER
-                        ,
-                        rfq.modemConfig.isHighPowerModule
-#endif
-  );
+  int16_t result = rfquack_rf->setOutputPower(rfq.modemConfig.txPower);
+  RFQUACK_LOG_TRACE("Output Power :     %f dBm (?), resultCode=%d", rfq.modemConfig.txPower, result)
 }
 
 /**
  * @brief Chages modem config choice in the radio driver
  */
 void rfquack_update_modem_config_choice() {
```

Comment:
Feel free to remove anything that has to do with RadioHead, including this canned config crap :-)